angr/knowledge_plugins/cfg/cfg_model.py

```diff
@@ -143,18 +147,28 @@ def serialize_to_cmessage(self):
 
     @classmethod
     def parse_from_cmessage(cls, cmsg, cfg_manager=None):  # pylint:disable=arguments-differ
-        model = cls(cmsg.ident, cfg_manager=cfg_manager)
+        if cfg_manager is None:
+            # create a new model unassociated from any project
+            model = cls(cmsg.ident)
+        else:
+            model = cfg_manager.new_model(cmsg.ident)
+
         # nodes
         for node_pb2 in cmsg.nodes:
-            node = CFGNode.parse_from_cmessage(node_pb2)
+            node = CFGNode.parse_from_cmessage(node_pb2, cfg=model)
             model._nodes[node.block_id] = node
-            model._nodes_by_addr[node.addr] = node
+            model._nodes_by_addr[node.addr].append(node)
             model.graph.add_node(node)
+            if len(model._nodes_by_addr[node.block_id]) > 1:
+                if once("cfg_model_parse_from_cmessage many nodes at addr"):
+                    l.warning("Importing a CFG with more than one node for a given address is currently unsupported. " +
+                              "The resulting graph may be broken.")
```

Comment:
```
LINT FAILURE: angr/knowledge_plugins/cfg/cfg_model.py regressed to 9.95/10.00
... angr/knowledge_plugins/cfg/cfg_model.py:164:20: W1201: Specify string format arguments as logging function parameters (logging-not-lazy)
```

You can remove the `+`. Multi-line strings are automatically put together as a long string.

Fix this and I'll hit merge!

Reply:
Whoops, should have linted before submitting.